contrib/mesos/pkg/executor/executor.go

```diff
@@ -162,24 +145,53 @@ func New(config Config) *Executor {
 		suicideWatch:         &suicideTimer{},
 		shutdownAlert:        config.ShutdownAlert,
 		exitFunc:             config.ExitFunc,
-		podStatusFunc:        config.PodStatusFunc,
 		staticPodsConfigPath: config.StaticPodsConfigPath,
 		launchGracePeriod:    launchGracePeriod,
 		nodeInfos:            config.NodeInfos,
 		initCompleted:        make(chan struct{}),
+		registry:             config.Registry,
+		kubeAPI:              &clientAPIWrapper{config.APIClient},
+		nodeAPI:              &clientAPIWrapper{config.APIClient},
 	}
+
 	runtime.On(k.initCompleted, k.runSendLoop)
 
-	po := newPodObserver(config.PodLW, k.updateTask, k.terminate)
-	runtime.On(k.initCompleted, po.run)
+	k.podWatcher = newPodWatcher(k.registry.watch())
```

Comment:
is this really the only place where a watch is created on the registry? If yes, it feels horribly over-engineered to have the broadcaster with its slots and the ringbuffer behind that. Maybe I am missing something.